src/scripts/alleles/update_alleles_mhcflurry.py

```diff
@@ -0,0 +1,580 @@
+    Iterates through JSON metadata to find missing loci, build rows for tsv using this data
+    """
+    new_loci_rows = set()
+    current_loci = set(current_loci)
+    missing_loci = set()
+
+    for species in metadata["species"]:
+        mhc_code = species["mhc_code"]
+        taxon = species["taxon"]
+
+        # Class I loci
+        for classI in species["class_I"]:
+            allele = mhc_code + "-" + classI
+            if "Eqca" in allele:
+                allele.replace("Eqca", "ELA")
+            if allele not in current_loci:
+                missing_loci.add(allele)
+                mhc_class = "MHC class I"
+                label = f"{allele} locus"
+                synonyms = ""
+                class_type = "subclass"
+                parent = f"{taxon} {mhc_class} locus"
+                new_loci_rows.add((label, synonyms, class_type, parent, ""))
+
+        # Class II loci
+        for classII in species["pairing"].keys():
+            allele1 = mhc_code + "-" + classII
+            allele2 = mhc_code + "-" + species["pairing"][classII]
+            if allele1 not in current_loci:
+                if "Eqca" in allele:
+                    allele1.replace("Eqca", "ELA")
+                missing_loci.add(allele1)
+                mhc_class = "MHC class II"
+                label = f"{allele1} locus"
+                synonyms = ""
+                class_type = "subclass"
+                parent = f"{taxon} {mhc_class} locus"
+                new_loci_rows.add((label, synonyms, class_type, parent, ""))
+            if allele2 not in current_loci:
+                if "Eqca" in allele2:
+                    allele.replace("Eqca", "ELA")
+                missing_loci.add(allele2)
+                mhc_class = "MHC class II"
+                label = f"{allele2} locus"
+                synonyms = ""
+                class_type = "subclass"
+                parent = f"{taxon} {mhc_class} locus"
+                new_loci_rows.add((label, synonyms, class_type, parent, ""))
+
+    with open(sys.argv[4], "a+") as outfile:
+        writer = csv.writer(outfile, delimiter="\t", lineterminator="\n")
+        for tup in new_loci_rows:
+            writer.writerow(tup)
+
+    return missing_loci
+
+
+def update_chains(curr_loci, ipd_seqs, allele_map, metadata, missing_loci):
+    """Updates the chain-sequence.tsv and chain.tsv with missing alleles from IPD
+
+    Returns:
+        A list of missing alleles ['A*02:01', 'B*02:01']
+    """
+
+    mro_alleles = set()
+    mro_gen_chains = set()
+    with open(sys.argv[2]) as fh:
+        rows = csv.DictReader(fh, delimiter="\t")
+        for row in rows:
+            allele = row["Label"].split(" ")[0]
+            allele = (":").join(allele.split(":")[:2])
+            mro_alleles.add(allele)
+            if "*" not in row["Label"]:
+                mro_gen_chains.add(row["Label"])
+
+    # Load MHCFlurry alleles
+    missing_alleles = set(ipd_seqs.keys()).difference(mro_alleles)
```

Comment:
I don't get this part. The `ipd_seqs.keys()` are accessions like "NHP00001" while the `mro_alleles` are like "HLA-B*51:151". So I get `len(ipd_seqs.keys())` = `len(missing_alleles)`.

Reply:
The idea is to get a list of alleles which are available in IPD-MHC, but aren't yet in MRO. I might have gotten it backwards but I was pretty sure that the accessions are the values in this script and the alleles are the keys. This script though is a one off that I wrote strictly for the tools team requirements and isn't really meant to be run beyond this PR. I have 2 scripts in progress that will scrape IPD-MHC and IMGT-HLA that I was planning on making a separate PR for. They're close to the script but much, much cleaner. I can remove this script from this PR for now though which is probably best?

Reply:
Ok. The tables look right to me and `make test` works. If we're not planning to use this script in the future then I won't worry about the details. We'll talk about this in our meeting tomorrow.